### DISCUSSION TITLE
TypeScript error with useInfiniteQuery

### DISCUSSION THREAD

**Discussion by Edjevw12** [TanStack/query#689] [General]
Updated react-query to the new types. Everything is working fine except useInfiniteQuery with key of array. Anyone know how to solve this one? 

![image](https://user-images.githubusercontent.com/37038043/86292917-ea431a00-bbf1-11ea-9823-cf23bbe1fd29.png)

**Comment by wolverineks** [comment]
could you provide a working example on codesandbox.io?

**Comment by wolverineks** [comment]
a good starting point
https://codesandbox.io/s/react-query-starter-8i7vb?file=/src/App.tsx

**Comment by qkreltms** [comment]
@Edjevw12 You can refer my example code to provide working example : https://codesandbox.io/s/react-query-useinfinitequery-be62c

**Reply by qkreltms** [reply]
I think this is happening because of third parameter of generic ```TMoreVariable```of useInfiniteQuery.
If you do not declare it, it passes type to queryFn as ```unknown``` type.
```ts
export type InfiniteQueryFunction<
  TResult,
  TKey extends AnyQueryKey,
  TMoreVariable
> = (
  ...keysAndMore: _.List.Append<TKey, TMoreVariable> | TKey
) => Promise<TResult>
```

Quick fix is define third generic type to any ```useInfiniteQuery<any, any, any, any>```
